src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -129,9 +159,25 @@
     lots of work when there are no inputs for code generation.
   -->
   <Target Name="ResolveRazorGenerateInputs">
+    <!--
+      In MVC Precompilation MvcRazorFilesToCompile also had the effect of suppressing the default
+      items for Razor code generation. As with all of these MVC Precompilation back-compat settings,
+      using the old thing, overrides the new thing.
+    -->
+    <PropertyGroup Condition="'@(MvcRazorFilesToCompile)'!=''">
+      <EnableDefaultRazorGenerateItems>false</EnableDefaultRazorGenerateItems>
+    </PropertyGroup>
     <ItemGroup>
-      <RazorGenerate Include="@(Content)" Condition="'%(Content.Extension)'=='.cshtml'">
-        <GeneratedOutput>$(RazorGenerateOutputPath)%(RelativeDir)%(Filename).cs</GeneratedOutput>
+      <RazorGenerate Include="@(MvcRazorFilesToCompile)" />
+    </ItemGroup>
+
+    <ItemGroup Condition="'$(EnableDefaultRazorGenerateItems)'=='true'">
+      <RazorGenerate Include="@(Content)" Condition="'%(Content.Extension)'=='.cshtml'" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <RazorGenerate Update="@(RazorGenerate)" Condtion="'%(RazorGenerate.GeneratedOutput)'==''">
```

Comment:
This change to use update is based on a bug we had in the existing code. If you added items to `RazorGenerate` before this change, they would get skipped because they don't have an output path.